tests/unit/test_charm.py

```diff
@@ -399,8 +344,36 @@ def test_db_relation(self):
             "database name should be set after relation joined",
         )
 
+    def test_add_admin_user(self):
+        self.harness.disable_hooks()
+        self.add_database_relations()
+
+        charm: DiscourseCharm = typing.cast(DiscourseCharm, self.harness.charm)
+        with patch.object(Container, "exec", return_value=MockExecProcess()) as exec_mock:
+            self.harness.container_pebble_ready("discourse")
+
+            email = "sample@email.com"
+            password = "somepassword"  # nosec
+            event = MagicMock()
```

Comment:
It is possible to supply a spec to Mocks using the `spec` parameter to ensure that all the interactions conform to a class or instance: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock